psycop/projects/t2d/paper_outputs/intervention_eval/hba1c.py

```diff
@@ -0,0 +1,79 @@
+import datetime as dt
+
+import pandas as pd
+import polars as pl
+
+from psycop.common.feature_generation.loaders.raw.load_lab_results import hba1c
+
+
+def time_from_first_pos_pred_to_next_hba1c(
+    pos_preds: pl.LazyFrame,
+    hba1cs: pl.LazyFrame,
+) -> pl.LazyFrame:
+    first_pos_pred_colname = "timestamp_first_pos_pred"
+    delta_time_col_name = "time_from_first_pos_pred_to_next_hba1c"
+
+    first_pos_pred = pos_preds.groupby("patient_id").agg(
+        pl.col("pred_timestamps").min().alias(first_pos_pred_colname),
+    )
+
+    filtered = (
+        first_pos_pred.join(
+            hba1cs,
+            on="patient_id",
+            how="inner",
+        )
+        .with_columns(
+            (pl.col("timestamp") - pl.col(first_pos_pred_colname)).alias(
+                delta_time_col_name,
+            ),
+        )
+        .filter(pl.col(delta_time_col_name) >= 0)
```

Comment:
what about test of this line? (prediction times before first tp should be excluded)